xds/src/main/java/io/grpc/xds/XdsServerBuilder.java

```diff
@@ -136,6 +137,15 @@ XdsServerBuilder xdsClientPoolFactory(XdsClientPoolFactory xdsClientPoolFactory)
     return this;
   }
 
+  /**
+   * Allows providing bootstrap override, useful for testing.
+   */
+  public XdsServerBuilder overrideBootstrapForTest(Map<String, ?> bootstrapOverride) {
+    this.xdsClientPoolFactory.setBootstrapOverride(
```

Comment:
why we can't?

Reply:
That means it would change channels and other servers. We went through this before on client-side. https://github.com/grpc/grpc-java/pull/8358/files/8f38f34705f34db27e9103928dc27dd856dc467e#r684332029